pkg/providers/tinkerbell/upgrade.go

```diff
@@ -443,20 +448,26 @@ func (p *Provider) handleRufioUnreleasedCRDs(ctx context.Context, cluster *types
 		return fmt.Errorf("retrieving hardware resources: %v", err)
 	}
 
-	for _, h := range hardware {
```

Comment:
These changes reduce updating Hardware to only apply if there is a diff. I anticipate updating this again with a different client call that filters for resources with non-nil BMC references eliminating most of this logic.